### ISSUE TITLE
Add support for writing OCSP requests and parsing OCSP responses

### ISSUE THREAD

**Issue by NWilson** [ARMmbed/mbedtls#1197]
### Description
- Type: Feature Request

This would be very useful, since CRLs are becoming less used and for Enterprise environments are simply unmanageable (we have customers with CRLs that are >25MiB).

The basic support for reading/writing OCSP requests and responses would be enough for us, since we integrate the revocation checks manually.

This would lay the groundwork for support of the OCSP stapling extension in TLS.

Note that I have implemented OCSP parsing/writing already, since we've waited for a couple of years for upstream to support it. I _could_ submit a PR - but I have a feeling it's might just languish in the PR backlog :( I can send you the code we have on request.

**Comment by RonEld** [comment]
Hi @NWilson Thank you for your Suggestion. I believe your request is a subset of #880 (although your issue is much more documented)

**Comment by NWilson** [comment]
Yes, it's a useful subset, and would be worth releasing on its own without sound the next stages of implementing OCSP stapling.

**Comment by khaefner** [comment]
@NWilson How far along did you get with OCSP?  Do you know of a fork that does support it?
Thanks!

Kyle

**Comment by NWilson** [comment]
Hi, I'm sorry I can't share my OCSP code (written for employer). OCSP stapling would be a little bit of extra work on top again. I agree it would be a highly desirable mbedTLS feature.
